afs/afs-core/src/main/java/com/powsybl/afs/Project.java

```diff
@@ -56,4 +56,10 @@ ProjectFile createProjectFile(NodeInfo nodeInfo) {
             return new UnknownProjectFile(context);
         }
     }
+
+    ProjectFolder createProjectFolder(NodeInfo nodeInfo) {
+        Objects.requireNonNull(nodeInfo);
```

Comment:
As you do not use nodeInfo directly, you should not assert `nodeInfo` is not null here. Could you check the assertion is done in `ProjectFileCreationContext` constructor and remove this line if the test is done.